tests/helics/system_tests/brokerTimeoutTests.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright (c) 2017-2019,
+Battelle Memorial Institute; Lawrence Livermore National Security, LLC; Alliance for Sustainable Energy, LLC.  See
+the top-level NOTICE for additional details. All rights reserved. SPDX-License-Identifier: BSD-3-Clause
+*/
+
+#include "../application_api/testFixtures.hpp"
+#include "gtest/gtest.h"
+#include "helics/core/core-exceptions.hpp"
+#include <complex>
+
+/** these test cases test out the value converters
+ */
+#include "helics/application_api.hpp"
+#include "helics/core/test/TestComms.h"
+#include "helics/core/test/TestCore.h"
+#include <future>
+
+#define CORE_TYPE_TO_TEST helics::core_type::TEST
+
+TEST (broker_timeout_tests, core_fail_timeout)
+{
+    auto brk = helics::BrokerFactory::create (CORE_TYPE_TO_TEST, "--timeout=200ms --tick 50ms");
+    brk->connect ();
+
+    helics::FederateInfo fi (CORE_TYPE_TO_TEST);
+    fi.coreName = "c1";
+
+    auto Fed1 = std::make_shared<helics::Federate> ("test1", fi);
+
+    fi.coreName = "c2";
+    auto Fed2 = std::make_shared<helics::Federate> ("test2", fi);
+
+    Fed1->enterExecutingModeAsync ();
+    Fed2->enterExecutingMode ();
+    Fed1->enterExecutingModeComplete ();
+
+    Fed1->finalize ();
+
+    auto cr = Fed2->getCorePointer ();
+    auto tcr = std::dynamic_pointer_cast<helics::testcore::TestCore> (cr);
+    EXPECT_TRUE (tcr);
+
+    auto cms = tcr->getCommsObjectPointer ();
+    cms->haltComms ();  // this will terminate communications abruptly
+    tcr.reset ();
+
+    bool val = brk->waitForDisconnect (std::chrono::milliseconds (2000));
```

Comment:
Does the time it takes to disconnect after the timeout much larger than 200ms, or does it vary significantly?

Reply:
It seems to vary quite a bit.  The tick only gets passed if no data has been sent, and that only occurs on the tick timeouts which rely on an independent thread operating.  Then it has to send the pings and wait for a sufficient number of ticks to pass, so the real timeout time is quite a bit longer than the timeout specified.  There are probably things we can do to refine it but it also is limited to number of threads on a CI system.  For it to function as a tick it really has to be independently operating and I don't want it so frequent that it accounts for a decent amount of processing.    I think we will refine it as we get more experience in it.  But I think this PR handles some of the most common scenarios with timeouts.